### PR TITLE
SCC-1318 | agent-client-auth 404s

### PR DESCRIPTION
**Pull request by Neozxc**

### CODE REVIEW

File: app/controllers/actions/AgentClientRedirectAction.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    for {
+      hasAgent <- request.userData.get(AgentClientQuery) match {
+                         case Some(v) =>
+                           logger.debug(s"Found existing agent client status for ${request.nino}")
+                           Future.successful(Right(v))
+                         case _       =>
+                           logger.debug(s"No agent client status for ${request.nino}, retrieving")
+                           callAgentClient
+                       }
+    } yield (request.uri.contains(config.manageTaxAgentsUrl), hasAgent) match {
```
Comment (owen-opencast) [inline app/controllers/actions/AgentClientRedirectAction.scala:75]: I think here we just need to yield the result - either true or false, including false if there's been an error.

Comment (owen-opencast) [reply app/controllers/actions/AgentClientRedirectAction.scala:75]: Move this logic to getSettingsSummary

File: app/controllers/IndexController.scala [hunk 2]
```diff
@@ -52,13 +52,14 @@ class IndexController @Inject() (
...
-  preferencesFrontendConnector: PreferencesFrontendConnector
+  preferencesFrontendConnector: PreferencesFrontendConnector,
+  agentClientCheck: AgentClientRedirectAction,
 )(implicit val ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with Logging {
 
-  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData andThen agentClientCheck).async { implicit request =>
```
Comment (owen-opencast) [inline app/controllers/IndexController.scala:62]: Remove action

Comment (owen-opencast) [reply app/controllers/IndexController.scala:62]: Remove call to agentClientAuthService on line 79

**Review by owen-opencast — CHANGES_REQUESTED** [review]

**Review by ccatosdevelopment — DISMISSED** [review]

**Review by owen-opencast — APPROVED** [review]

**Pull request merged**